model/generate.go

```diff
@@ -199,8 +199,8 @@ func cacheProjectData(p *Project) projectMaps {
 	for _, bv := range p.BuildVariants {
 		cachedProject.buildVariants[bv.Name] = struct{}{}
 	}
-	for _, t := range p.Tasks {
-		cachedProject.tasks[t.Name] = &t
+	for i, t := range p.Tasks {
+		cachedProject.tasks[t.Name] = &p.Tasks[i]
```

Comment:
should we add a comment to this or do we feel comfortable enough given the test

Reply:
We assign local variables in a lot of places, so I don't know if it's worth putting a comment specifically here when it happens all over the place. Maybe we can advertise a list of common Go gotchas during a team presentation to explain why this is correct and the previous thing isn't?

Reply:
i think a team presentation isn't really necessary since the issue with gotchas is that they're easy to forget unless you run into them (although could still be a fun one so i'm not against having one haha). i'm fine not leaving a comment though since it is tested

Reply:
Maybe we can compile a list of common bugs/practices to avoid in Evergreen with a Google doc, and if we see other gotchas come up in other code reviews, we could add to the list?